packages/firestore/lib/modular/index.d.ts

```diff
@@ -0,0 +1,546 @@
+ *
+ * In this use case, `V` is used to distribute the union types of `T[K]` on Record, since `T[K]`
+ * is evaluated as an expression and not distributed.
+ *
+ * See https://www.typescriptlang.org/docs/handbook/advanced-types.html#distributive-conditional-types
+ */
+export declare type ChildUpdateFields<K extends string, V> = V extends Record<string, unknown>
+  ? AddPrefixToKeys<K, UpdateData<V>>
+  : never;
+
+/**
+ * For each field (e.g. 'bar'), find all nested keys (e.g. {'bar.baz': T1, 'bar.qux': T2}).
+ * Intersect them together to make a single map containing all possible keys that are all marked as optional
+ */
+export declare type NestedUpdateFields<T extends Record<string, unknown>> = UnionToIntersection<
+  {
+    [K in keyof T & string]: ChildUpdateFields<K, T[K]>;
+  }[keyof T & string]
+>;
+
+/**
+ * Update data (for use with {@link updateDoc}) that consists of field paths (e.g. 'foo' or 'foo.baz')
+ * mapped to values. Fields that contain dots reference nested fields within the document.
+ * FieldValues can be passed in as property values.
+ */
+export declare type UpdateData<T> = T extends Primitive
+  ? T
+  : T extends object
+  ? {
+      [K in keyof T]?: UpdateData<T[K]> | FieldValue;
+    } & NestedUpdateFields<T>
+  : Partial<T>;
+
+/**
+ * Allows FieldValues to be passed in as a property value while maintaining
+ * type safety.
+ */
+export type WithFieldValue<T> =
+  | T
+  | (T extends Primitive
+      ? T
+      : T extends object
+      ? { [K in keyof T]: WithFieldValue<T[K]> | FieldValue }
+      : never);
+
+/**
+ * Returns the existing default {@link Firestore} instance that is associated with the
+ * provided {@link @firebase/app#FirebaseApp}. If no instance exists, initializes a new
+ * instance with default settings.
+ *
+ * @param app - The {@link @firebase/app#FirebaseApp} instance that the returned {@link Firestore}
+ * instance is associated with.
+ * @returns The {@link Firestore} instance of the provided app.
+ */
+export function getFirestore(app: FirebaseApp): Firestore;
+
+/**
+ * Returns the existing default {@link Firestore} instance that is associated with the
+ * default {@link @firebase/app#FirebaseApp}. If no instance exists, initializes a new
+ * instance with default settings.
+ *
+ * @returns The {@link Firestore} instance of the provided app.
+ */
+export declare function getFirestore(): Firestore;
+
+/**
+ * Returns the existing default {@link Firestore} instance that is associated with the
+ * provided {@link @firebase/app#FirebaseApp}. If no instance exists, initializes a new
+ * instance with default settings.
+ *
+ * @param app - The {@link @firebase/app#FirebaseApp} instance that the returned {@link Firestore}
+ * instance is associated with.
+ * @returns The {@link Firestore} instance of the provided app.
+ * @internal
+ */
+export declare function getFirestore(app: FirebaseApp): Firestore;
```

Comment:
is this duplicate?  https://github.com/invertase/react-native-firebase/pull/7235/files#diff-a6ee9351de272b709cac1ff7ca3c24f2db30fe7185a0791fb9ac3c9db6b88c34R99-R107